Kernel/KBO.cpp

```diff
@@ -57,11 +59,18 @@ class KBO::State
     _varDiffs.reset();
   }
 
-  void traverse(Term* t1, Term* t2);
-  void traverse(TermList tl,int coefficient);
-  Result result(Term* t1, Term* t2);
-private:
-  void recordVariable(unsigned var, int coef);
+  template<bool unidirectional>
+  Result traverseLex(AppliedTerm t1, AppliedTerm t2);
+  template<bool unidirectional>
+  Result traverseNonLex(AppliedTerm t1, AppliedTerm t2);
```

Comment:
Maybe add a comment like: `traverseLex` is the "normal" traversing as in the paper <title>, and `traverseNonLex` is an optimized version of `traverseLex` that gives the same results as `traverseLex` but does not compute the result `_lexResult` and can be used in cases where we only care about the weight difference and the variable result, but not the lex result. 
(I don't think what I wrote here is quite accurate but a correct comment in the same spirit would be sweet )